src/hugr/views/petgraph.rs

```diff
@@ -1,187 +1,250 @@
 //! Implementations of petgraph's traits for Hugr Region views.
 
-use super::{DescendantsGraph, SiblingGraph};
-use crate::hugr::views::sealed::HugrInternals;
 use crate::hugr::HugrView;
-use crate::ops::handle::NodeHandle;
 use crate::ops::OpType;
 use crate::types::EdgeKind;
 use crate::{Node, Port};
 
 use context_iterators::{ContextIterator, IntoContextIterator, MapWithCtx};
+use delegate::delegate;
 use petgraph::visit as pv;
 use portgraph::NodeIndex;
 
-macro_rules! impl_region_petgraph_traits {
-    ($hugr:ident) => {
-        impl<'a, Root, Base> pv::GraphBase for $hugr<'a, Root, Base>
-        where
-            Root: NodeHandle,
-            Base: HugrInternals + HugrView,
-        {
-            type NodeId = Node;
-            type EdgeId = ((Node, Port), (Node, Port));
-        }
+use super::sealed::HugrInternals;
 
-        impl<'a, Root, Base> pv::GraphProp for $hugr<'a, Root, Base>
-        where
-            Root: NodeHandle,
-            Base: HugrInternals + HugrView,
-        {
-            type EdgeType = petgraph::Directed;
-        }
+/// Wrapper for a HugrView that implements petgraph's traits.
+///
+/// It can be used to apply petgraph's algorithms to a Hugr.
+#[derive(Debug, Clone, Copy)]
+pub struct PetgraphWrapper<'a, T> {
+    hugr: &'a T,
+}
 
-        impl<'a, Root, Base> pv::NodeCount for $hugr<'a, Root, Base>
-        where
-            Root: NodeHandle,
-            Base: HugrInternals + HugrView,
-        {
-            fn node_count(&self) -> usize {
-                HugrView::node_count(self)
-            }
-        }
+impl<'a, T> PetgraphWrapper<'a, T>
+where
+    T: HugrView,
+{
+    /// Wrap a HugrView in a PetgraphWrapper.
```

Comment:
I guess this is good to make sure you don't construct useless PetgraphWrappers (containing other things). But, given PetgraphWrapper's only field is hidden, so they are not publically constructable - how about making the "public constructor" be an `fn as_petgraph(&self)` in <strike>HierarchyView/</strike>HugrView?

Reply:
The bound here is just so we get a nice "Missing HugrView" error here instead of "Petgraph trait not implemented, here's some options".

`as_petgraph` would be handy 👍

Reply:
Given we have `as_petgraph` now, indeed on HugrView rather than on HierarchyView :-), do we still need this?